src/wolfkrow/core/engine/resolver.py

```diff
@@ -155,17 +291,19 @@ def _replace_replacements(self, value):
                 value = value.replace(sub, template_value)
 
         # Finally replace any extra environment variables which were added from the 
-        # values of replaceed replacements.
+        # values of replaced replacements.
         value = os.path.expandvars(value)
 
         return value
         
-    def _get_sgtk_template_value(self, template_name):
+    def _get_sgtk_template_value(self, template_name, replacements=None):
         """ Will attempt to find the given template name in the sgtk instace, and 
             then use the replacements dict as the fields to get the substituted
             template.
         """
 
+        replacements = replacements or self.resolved_replacements
+
         if self.sgtk is None:
             # TODO: ERROR: Workflow builder initialized without sgtk instance, 
             # BUT sgtk templates found in workflow.
```

Comment:
We can actually raise an error here now because the resolver will only try to resolver SGTK templates which are actually used.